cmake/extensions.cmake

```diff
@@ -1960,6 +1960,47 @@ Relative paths are only allowed with `-D${ARGV1}=<path>`")
   endif()
 endfunction()
 
+# Usage:
+#   zephyr_string(<mode> <out-var> <input> ...)
+#
+# Zephyr string function extension.
+# This function extends the CMake string function by providing additional
+# manipulation arguments to CMake string.
+#
+# SANITIZE: Ensure that the output string does not contain any special
+#           characters. Special characters, such as -, +, =, $, etc. are
+#           converted to underscores '_'.
+#
+# SANITIZE TOUPPER: Ensure that the output string does not contain any special
+#                   characters. Special characters, such as -, +, =, $, etc. are
+#                   converted to underscores '_'.
```

Comment:
@tejlmand what is the actual impact of this conversion? No objections, but would like to know if there are modules that actually have special characters in the names :)

Reply:
Yes, we have modules with `-` in their names. But the path won't change with this PR, just the internal name used by CMake and Kconfig.

Reply:
as we now introduces the module as Kconfig variable for proper glue code handling outside of the module, then we need to sanitize the name, as it goes into `autoconf`, like this:
```
.../build/zephyr/include/generated/autoconf.h:37:9: warning: ISO C99 requires whitespace after the macro name
   37 | #define CONFIG_ZEPHYR_TRUSTED-FIRMWARE-M_MODULE 1
```
we could of course just sanitize for Kconfig symbols, but I would like that all places are using same form, so therefore sanitizing everywhere.